source/CScriptEngine.cpp

```diff
@@ -1,1038 +1,1073 @@
+        std::for_each(CustomScripts.begin(), CustomScripts.end(), [this](CCustomScript *cs) {
+            RemoveScriptFromQueue(cs, activeThreadQueue);
+            cs->SetActive(false);
+        });
+    }
+
+    void CScriptEngine::ReregisterAllScripts()
+    {
+        TRACE("Reregistering all custom scripts");
+        std::for_each(CustomScripts.begin(), CustomScripts.end(), [this](CCustomScript *cs) {
+            AddScriptToQueue(cs, activeThreadQueue);
+            cs->SetActive(true);
+        });
+    }
+
+    CCustomScript::CCustomScript(const char *szFileName, bool bIsMiss, CCustomScript *parent, int label)
+        : CRunningScript(), bSaveEnabled(false), bOK(false),
+        LastSearchPed(0), LastSearchCar(0), LastSearchObj(0),
+        CompatVer(CLEO_VERSION)
+    {
+        IsCustom(1);
+        bIsMission = bUseMissionCleanup = bIsMiss;
+        UseTextCommands = 0;
+        NumDraws = 0;
+        NumTexts = 0;
+
+        TRACE("Loading custom script %s...", szFileName);
+
+        try
+        {
+			std::ifstream is;
+			if (label != 0) // Create external from label.
+			{
+				if (!parent)
+					throw std::logic_error("Trying to create external thread from label without parent thread");
+				BaseIP = parent->GetBasePointer();
+				CurrentIP = parent->GetBasePointer() - label;
+				memcpy(Name, parent->Name, sizeof(Name));
+				dwChecksum = parent->dwChecksum;
+				parentThread = parent;
+				parent->childThreads.push_back(this);
+			}
+			else
+			{
+				using std::ios;
+				std::ifstream is(szFileName, std::ios::binary);
+				is.exceptions(std::ios::badbit | std::ios::failbit);
+				std::size_t length;
+				is.seekg(0, std::ios::end);
+				length = is.tellg();
+				is.seekg(0, std::ios::beg);
+
+				if (bIsMiss)
+				{
+					if (*MissionLoaded)
+						throw std::logic_error("Starting of custom mission when other mission loaded");
+					*MissionLoaded = 1;
+					BaseIP = CurrentIP = missionBlock;
+				}
+				else {
+					BaseIP = CurrentIP = new BYTE[length];
+				}
+				is.read(reinterpret_cast<char *>(BaseIP), length);
+
+				auto fname = strrchr(szFileName, '\\') + 1;
+				if (!fname) fname = strrchr(szFileName, '/') + 1;
+				if (fname < szFileName) fname = szFileName;
+				memcpy(Name, fname, sizeof(Name));
+				Name[7] = '\0';
+				dwChecksum = crc32(reinterpret_cast<BYTE *>(BaseIP), length);
+			}
+			lastScriptCreated = this;
+            bOK = true;
+        }
+        catch (std::exception& e)
+        {
+            TRACE("Error during loading of custom script %s occured.\nError message: %s", szFileName, e.what());
+        }
+        catch (...)
+        {
+            TRACE("Unknown error during loading of custom script %s occured.", szFileName);
+        }
+    }
+
+    CCustomScript::~CCustomScript()
+    {
+        if (BaseIP && !bIsMission) delete[] BaseIP;
+		RunScriptDeleteDelegate(reinterpret_cast<CRunningScript*>(this));
+		if (lastScriptCreated == this) lastScriptCreated = nullptr;

```

Comment:
not sure how `lastScriptCreated` is used (I see it exported in API) but I would expect to point at the second to last script created. i.e.:
```
create script A // lastScriptCreated = A
create script B // lastScriptCreated = B
end script B // lastScriptCreated = A
end script A // lastScriptCreated = null
```

Reply:
Yes, it could improve, I just don't think it's so important, so, whatever.

In my idea, the usage was: `STREAM_CUSTOM_SCRIPT` does not return the pointer of the thread that was just created, so, shortly after calling it, a command that returns the pointer of the last created thread.
I put this in the CLEO + plugin:

    STREAM_CUSTOM_SCRIPT_FROM_LABEL Test47
    GET_LAST_CREATED_CUSTOM_SCRIPT i

I only set it to null if this script is deleted to make it safer to not return a deleted thread, it is not expected to be used outside the case I just exemplified now (at least I don't know of any other use). So I found it unnecessary to store a sequence of scripts or something, too overkill.

Reply:
Gotcha